src/core/test/system/borrower/unlock.t.sol

```diff
@@ -0,0 +1,162 @@
+// Copyright (C) 2019 Centrifuge
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.12;
+
+import "../system.sol";
+import "../users/borrower.sol";
+
+contract UnlockTest is SystemTest {
```

Comment:
I think we should test the case, where the NFT is locked in the collector

Reply:
yes, there is a todo in the cases for it. Coming in the PR with collector tests :)